go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -3811,6 +3853,69 @@ func (s *VtctldServer) ValidateVSchema(ctx context.Context, req *vtctldatapb.Val
 	return &resp, nil
 }
 
+// getTopologyCell gets a cell and its children recursively, given a cell's path
+func (s *VtctldServer) getTopologyCell(ctx context.Context, cellPath string) (*vtctldatapb.TopologyCell, error) {
+	// Extract cell and relative path
+	parts := strings.Split(cellPath, "/")
+	if parts[0] != "" || len(parts) < 2 {
+		return nil, fmt.Errorf("Invalid path: %s", cellPath)
+	}
+	cell := parts[1]
+	relativePath := cellPath[len(cell)+1:]
```

Comment:
you don't need to do this math if you use `strings.SplitN`: https://go.dev/play/p/CZXhsrV0bUj

Reply:
I don't mind leaving it as the cellPath as opposed to `relativePath := strings.SplitN(cellPath, "/", 3)[1]` - just because personally, array slicing is more easily understood.

Reply:
works for me!